apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -39,18 +71,33 @@ class PageTemplateModal extends Component {
 	);
 
 	// Parse templates blocks and memoize them.
-	getBlocksByTemplateSlugs = memoize( templates =>
-		reduce(
+	getBlocksByTemplateSlugs = memoize( templates => {
+		return reduce(
 			templates,
 			( prev, { slug, content } ) => {
-				prev[ slug ] = content
-					? parseBlocks( replacePlaceholders( content, this.props.siteInformation ) )
-					: [];
+				if ( ! content ) {
+					prev[ slug ] = [];
+					return prev;
+				}
+
+				let blocks;
+
+				// Replacements on raw Block Grammar
+				blocks = replacePlaceholders( content, this.props.siteInformation );
+
+				// Parse the Blocks
+				blocks = parseBlocks( blocks );
+
+				// Replacements on Parsed Blocks
+				// blocks = modifyParsedBlocks( blocks );
```

Comment:
I removed this as I realised it modified the root blocks used everywhere where we only want to modify the blocks passed to the preview.

However I'm thining in a followup PR we could introduce filter hooks here to allow global modification of all blocks. If not then all the amends introduce around this area can be reverted.